test/server/auth-test.js

```diff
@@ -0,0 +1,134 @@
+var expect       = require('chai').expect;
+var express      = require('express');
+var ntlm         = require('express-ntlm');
+var auth         = require('basic-auth');
+var request      = require('request');
+var Proxy        = require('../../lib/proxy');
+var Session      = require('../../lib/session');
+var requestAgent = require('../../lib/request-pipeline/destination-request/agent');
+
+describe('Authentication', function () {
+    var proxy   = null;
+    var session = null;
+
+    beforeEach(function () {
+        session = new Session();
+
+        session.getAuthCredentials = function () {
+            return null;
+        };
+
+        session.handleFileDownload = function () {
+        };
+
+        proxy = new Proxy('127.0.0.1', 1836, 1837);
+    });
+
+    afterEach(function () {
+        proxy.close();
+        requestAgent.resetKeepAliveConnections();
+    });
+
+    describe('NTLM Authentication', function () {
+        var ntlmServer = null;
+
+        before(function () {
+            var app = express();
+
+            app.use(ntlm());
+
+            app.all('*', function (req, res) {
+                res.end(JSON.stringify(req.ntlm));
+            });
+
+            ntlmServer = app.listen(1506);
+        });
+
+        after(function () {
+            ntlmServer.close();
+        });
+
+
+        it('Should authorize with correct credentials', function (done) {
```

Comment:
Unfortunately, this is difficult to do, we need a full `NTML` server, I could not do it with the `express-ntlm`